vaults/sources/Vault.move

```diff
@@ -1,80 +1,125 @@
 module titusvaults::Vault {
     use aptos_framework::coin::{Self, Coin};
     use std::signer;
-    use std::smart_table::{Self, SmartTable};
+    use std::signer::address_of;
+    use aptos_framework::account;
+    use aptos_std::smart_table::{Self, SmartTable};
+    use aptos_framework::randomness::u64_integer;
+    use aptos_framework::timestamp;
 
-    const E_INVALID_OPERATION: u64 = 69;
+    /// When signer is not owner of module
+    const E_NOT_AUTHORIZED: u64 = 1;
 
+    const E_INVALID_OPERATION: u64 = 69;
 
-    struct Vault <phantom CollateralType> has key {
-        deposits: SmartTable<address, DepositBalance<CollateralType>>,
+    /// each users have there own vault modules
```

Comment:
this is probably not the right mental model

each STRATEGY has their own vault, with a set of users depositing into each